keras_cv/layers/preprocessing/erasing.py

```diff
@@ -0,0 +1,226 @@
+    """
+
+    def __init__(self, rate, patch_value=None, seed=None, **kwargs):
+        super().__init__(**kwargs)
+        self.rate = rate
+        self.patch_value = patch_value
+        self.seed = seed
+
+    def call(self, images, labels):
+        """call method for the layer.
+
+        Args:
+            images: Tensor representing images of shape [batch_size, width, height, channels], with dtype tf.float32.
+            labels: original labels.
+        Returns:
+            images: augmented images, same shape as input.
+            labels: orignal labels.
+        """
+
+        augment_cond = tf.less(
+            tf.random.uniform(shape=[], minval=0.0, maxval=1.0), self.rate
+        )
+        # pylint: disable=g-long-lambda
+        random_erase_augment = lambda: self._erase(images, labels)
+        no_augment = lambda: (images, labels)
+        return tf.cond(augment_cond, random_erase_augment, no_augment)
+
+    def _erase(self, images, labels):
+        """Apply random erase."""
+        input_shape = tf.shape(images)
+        batch_size, image_height, image_width = (
+            input_shape[0],
+            input_shape[1],
+            input_shape[2],
+        )
+
+        cut_height, cut_width = self._compute_cut_size(
+            batch_size, image_height, image_width
+        )
+
+        random_center_height = tf.random.uniform(
+            shape=[batch_size], minval=0, maxval=image_height, dtype=tf.int32
+        )
+        random_center_width = tf.random.uniform(
+            shape=[batch_size], minval=0, maxval=image_width, dtype=tf.int32
+        )
+
+        args = [
+            images,
+            random_center_width,
+            random_center_height,
+            cut_width // 2,
+            cut_height // 2,
+        ]
+        if self.patch_value is not None:
+            patch_value = tf.fill([batch_size], self.patch_value)
+            args.append(patch_value)
+
+        images = tf.map_fn(
+            lambda x: _fill_rectangle(*x),
+            args,
+            fn_output_signature=tf.TensorSpec.from_tensor(images[0]),
+        )
+
+        return images, labels
+
+    def get_config(self):
+        config = {
+            "rate": self.rate,
+            "patch_value": self.patch_value,
+            "seed": self.seed,
+        }
+        base_config = super(BaseErasing, self).get_config()
+        return dict(list(base_config.items()) + list(config.items()))
+
+    @abc.abstractmethod
+    def _compute_cut_size(self, batch_size, image_height, image_width):
+        pass
+
+
+class RandomErasing(BaseErasing):
+    """RandomErasing implements the RandomErasing data augmentation technique.
+
+    Args:
+        rate: Float between 0 and 1.  The fraction of samples to augment.
+        scale: Tuple of float. Area scale range (min, max) of erasing patch.
+        ratio: Tuple of float. Aspect ratio range (min, max) of erasing patch.
+        patch_value: Float. The value to fill in the patches. If None, will
```

Comment:
Defaults to `None`.  Let's also give this a value instead of `None`.  Maybe `noise` or `random`.

Reply:
I have decided on `"gaussian"`.